### ISSUE TITLE
Rename the images you send by CTRL + V

### ISSUE THREAD

**Issue by Valdnet** [nextcloud/spreed#3543]
When sending files via drag'n'drop, I suggest changing the file name from e.g.: image (2) to the current date and time of sending (20200513_202433). In my opinion it will be more professional.

![obraz](https://user-images.githubusercontent.com/47037905/81850481-51c8db80-9558-11ea-8b1c-12f5126db9c5.png)

**Comment by vix95** [comment]
Hi, I think that it's a good idea and it looks more pro as well. But this one should refer only for images by CTRL + V. The real image file name is ok for me, if it's dragged&dropped from Desktop for instance.

**Comment by Valdnet** [comment]
> But this one should refer only for images pasted by CTRL + V.

I agree with you. After thorough checking of the sending of specific files, the names should be unchanged and a tag (2) added etc.

**Comment by PVince81** [comment]
When uploading more than 4 files, the dialog already looks crowded and file names are ellipsized already.
It wouldn't look ideal to have a pencil icon there to rename inline, nor would it look nice to have another modal on top for renaming.

The layout of that upload dialog might need to be reworked a bit.

**Comment by nickvergessen** [comment]
Just rename them with a timestamp

**Comment by PVince81** [comment]
I think it depends on the purpose of the file name within a conversation.

Usually the file name is always displayed within the chat. Some people might want the name to be like a description of the picture itself, or hide the name completely and post an own description in the next message. If that name cannot be changed I still wonder whether we should display it at all then.

Another use of the file name would be to distinguish one file from others, for example in a series of photos or documents where one wants to download the correct/intended one.

Always automatically renaming the file names would break the above use cases.
This is why intuitively I thought we should allow renaming the files within that dialog so that the name appears correctly and also on disk. Note: renaming the file on disk after the fact does not rename it for other users, so it has to be done before upload (and implicit share)

**Comment by PVince81** [comment]
PR here: https://github.com/nextcloud/spreed/pull/4485
